filters.im

```diff
@@ -1556,64 +1556,47 @@ i_rgbdiff_image(i_img *im1, i_img *im2) {
 
   out = i_sametype_chans(im1, xsize, ysize, outchans);
 
-  if (im1->bits == i_8_bits && im2->bits == i_8_bits) {
-    i_color *line1 = mymalloc(xsize * sizeof(*line1));
-    i_color *line2 = mymalloc(xsize * sizeof(*line1));
+#code im1->bits == i_8_bits && im2->bits == i_8_bits
+    IM_COLOR *line1 = mymalloc(xsize * sizeof(*line1));
+    IM_COLOR *line2 = mymalloc(xsize * sizeof(*line1));
     i_img_dim x, y;
     int ch;
+    IM_WORK_T opaque;
+#ifdef IM_EIGHT_BIT
+    opaque = 255;
+#else
+    opaque = 1.0;
+#endif
 
     for (y = 0; y < ysize; ++y) {
-      i_glin(im1, 0, xsize, y, line1);
-      i_glin(im2, 0, xsize, y, line2);
+      IM_GLIN(im1, 0, xsize, y, line1);
+      IM_GLIN(im2, 0, xsize, y, line2);
       if (outchans != diffchans) {
         /* give the output an alpha channel since it doesn't have one */
-        for (x = 0; x < xsize; ++x)
-          line2[x].channel[diffchans] = 255;
-      }
-      for (x = 0; x < xsize; ++x) {
-        for (ch = 0; ch < diffchans; ++ch) {
-          if (ch == outchans-1) {
-            line2[x].channel[ch] = 255; /* ignore alpha channel */
-          } else {
-            line2[x].channel[ch] = i_abs(line1[x].channel[ch] - line2[x].channel[ch]);
-          }
+        for (x = 0; x < xsize; ++x) {
+          line2[x].channel[diffchans] = opaque;
```

Comment:
opaque can just be IM_SAMPLE_MAX

Reply:
opaque is not needed anymore when we remove the alpha channel